### PR TITLE
Self-Referential Links Fix for Markdown View

### LINKED ISSUE #261: Self-referential links dont work in markdown view
Self-referential links (jumping to another section inside the same document) are not working when displaying the Github READMEs for models and datasets in the MLX UI (using the `npm` package [`react-markdown`](https://www.npmjs.com/package/react-markdown))

i.e.  https://ml-exchange.org/models/codenet-language-classification#deploy-on-red-hat-openshift

> ...
> ## Deployment Options
> <li><a href="#deploy-from-container-registry">Deploy from Container Registry</a></li>
> <li><a href="#deploy-on-red-hat-openshift">Deploy on Red Hat OpenShift</a></li>
> <li><a href="#deploy-on-kubernetes">Deploy on Kubernetes</a></li>
> <li><a href="#run-locally">Run Locally</a></li></ul>
> ...

I believe @drewbutlerbb4 had tried to make these kinds of internal links work, by exploring different `npm` libraries for rendering GitHub style markdowns. But I don’t remember where that led.

Absolute links that go outside of ml-exchange.org should work.

### PR DESCRIPTION
**Pull request by RRM123**
Upgraded / Added the following dependencies to `dashboard/origin-mlx/package.json`:

- `react-markdown`: ^4.0.8 --> ^8.0.0
- `remark-gfm` : ^2.0.0 --> ^3.0.0
- `rehype-slug` : ^5.0.1

Changes were made to `OperatorDetail.tsx`, `MarkdownViewer.tsx`, and `MetaCard.tsx` to 
accommodate the new version of `react-markdown` and `rehype-slug`.

Resolves #261

/assign @ckadner

### CODE REVIEW

**Comment by BluThaitanium** [comment]
Awesome work! /lgtm

File: dashboard/origin-mlx/package.json [hunk 1]
```diff
@@ -28,17 +28,19 @@
...
     "react-tooltip": "^4.2.8",
     "reactjs-popup": "^1.5.0",
     "reactstrap": "^8.5.1",
-    "remark-gfm": "^2.0.0",
+    "rehype-slug": "^5.0.1",
+    "remark-gfm": "^3.0.0",
     "styled-components": "^4.3.1",
     "title-case": "^3.0.3",
     "typescript": "^3.5.2",
+    "typescript-logging": "^1.0.0",
```
Comment (ckadner) [inline dashboard/origin-mlx/package.json:43]: I assume you added the `typescript-logging` dependency to help with debugging, but you removed any logging statements from the code in your PR. Did the logging not work well or did you find it was not necessary given other means of debugging?

**Review by ckadner — CHANGES_REQUESTED** [review]
Thanks @RRM123 -- looks good. Just a small question.

File: dashboard/origin-mlx/src/pages/MetaFeaturedPage/MetaCard.tsx [hunk 2]
```diff
@@ -97,16 +97,16 @@ function MetaCard(props: MetaCardProps) {
...
+                children={description} components={{
+                  p: ({children}) =>
                     <Typography
                       variant="body1"
                       component="p"
                       className={classes.description}
                     >
                       {children}
                     </Typography>, 
-                  link: ({children}) => children
```
Comment (ckadner) [inline dashboard/origin-mlx/src/pages/MetaFeaturedPage/MetaCard.tsx:109]: FYI, this link transformation rule did not seem to do anything, so we should be okay removing it, or, commenting it out to keep for future reference

File: dashboard/origin-mlx/package.json (hunk 1 above)
Comment (ckadner) [reply dashboard/origin-mlx/package.json]: I updated the `package.json` file to remove the `typescript-logging` dependency. 

I also regenerated the `package-lock.json` as the one you generated did not look right. Are you using an old version of NodeJS/`npm`?

**Review by ckadner — APPROVED** [review]
/lgtm

**Pull request merged**